### PR TITLE
fix: 10263 don't explode for bad option values

### PR DESCRIPTION
**Pull request by robinbryce**
Also expand the test coverage to fully cover verification modes.

Positive and negative cases for `--leaf`, `--entryid` and `--event-json-file`

Negative *tamper* case for `--event-json-file`

AB#10263

### CODE REVIEW

File: datatrails_scitt_samples/scripts/verify_receipt.py [hunk 1]
```diff
@@ -132,10 +145,13 @@ def main():
...
-        print("signature verification failed")
+        print("verification succeeded")
+        return True
+    print("verification failed")
+    return False
 
 
 if __name__ == "__main__":
-    main()
+    if not main():
```
Comment (eccles) [inline datatrails_scitt_samples/scripts/verify_receipt.py:155]: why not make main() return an int (0 or 1) then this could be one line

Comment (eccles) [reply datatrails_scitt_samples/scripts/verify_receipt.py:155]: sys.exit(main())

File: tests/test_verify_receipt.py [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+from datatrails_scitt_samples.scripts.verify_receipt import (
+    main as verify_receipt,
+)
+
+
+class TestVerifyReciept(unittest.TestCase):
+    """
+    Tests verification of a known receipt.
+    """
+
```
Comment (eccles) [inline tests/test_verify_receipt.py:37]: add line:

     maxDiff = None

and failures with diff with be output in sensible format.

https://docs.python.org/3/library/unittest.html#unittest.TestCase.maxDiff

File: tests/test_verify_receipt.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+        )
+        self.assertTrue(os.path.exists(f"{self.test_dir}/signed-statement.cbor"))
+
+        # register the signed statement
+        output = io.StringIO()
+        with redirect_stdout(output):
+            register_signed_statement(
+                [
+                    "--signed-statement-file",
+                    f"{self.test_dir}/signed-statement.cbor",
```
Comment (eccles) [inline tests/test_verify_receipt.py:89]: if composing filepaths use os.path.join

File: datatrails_scitt_samples/scripts/verify_receipt.py (hunk 1 above)
Comment (robinbryce) [reply datatrails_scitt_samples/scripts/verify_receipt.py:155]: that is what I originaly had. I thought the objection would be the otherway round - that returning ints from a "function" was fussy. I will switch it back

Comment (robinbryce) [reply datatrails_scitt_samples/scripts/verify_receipt.py:155]: note this means verified == 0 is *success* (i am fine with that)

File: tests/test_verify_receipt.py (hunk 3 above)
Comment (robinbryce) [reply tests/test_verify_receipt.py:89]: I really don't think that matters on any platform we support (including windows) but ok

**Review by eccles — APPROVED** [review]

**Pull request merged**